### DISCUSSION TITLE
When I call createParticipant, a lot of errors occur: No buffer space available

### DISCUSSION THREAD

**Discussion by WuGaoyin** [eProsima/Fast-DDS#3184] [Unofficial support]
### Is there an already existing issue for this?

- [X] I have searched the existing issues

### Expected behavior

The program can send and receive messages normally after calling createParticipant and createPublisher/createSubscriber

### Current behavior

The program cann‘t send and receive messages normally after calling createParticipant and createPublisher/createSubscriber.

To better investigate the cause, I turned on the rtps debug switch. Then it is found that when createParticipant is called, a large number of error messages(set_option: No buffer space available) are printed continuously. The detailed log is as follows:

[RTPS_PARTICIPANT Info]  -> Function createParticipant
[RTPS_PARTICIPANT Info] Thread: 4 created and waiting for tasks. -> Function announce_thread
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26000) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26002) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26004) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26006) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26008) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26010) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26012) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26014) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26016) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26018) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26020) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26022) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26024) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26026) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26028) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26030) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26032) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26034) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26036) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26038) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26040) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26042) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26044) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26046) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26048) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26050) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26052) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26054) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26056) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26058) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26060) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26062) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26064) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26066) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets
[RTPS_MSG_OUT Info] UDPv4 Error binding at port: (26068) with msg: set_option: No buffer space available -> Function OpenAndBindInputSockets

### Steps to reproduce

Just call createParticipant function.

### Fast DDS version/commit

FastRtps 1.5.0

### Platform/Architecture

Other. Please specify in Additional context section.

### Transport layer

UDPv4

### Additional context

OS: QNX7.0.0
Architecture: AArch64

**Comment by JLBuenoLopez** [comment]
Hi @WuGaoyin,

FastRTPS v1.5.0 is no longer supported (see [Fast DDS release support](https://github.com/eProsima/Fast-DDS/blob/master/RELEASE_SUPPORT.md)) and also QNX is not an officially supported platform (see [Fast DDS supported platforms](https://github.com/eProsima/Fast-DDS/blob/master/PLATFORM_SUPPORT.md)]. I suggest on the one hand that you use some supported version and maybe someone in the community can help you with QNX support.

In the meantime, this issue should be moved to the Unofficial Support Discussion forum according with [Fast DDS CONTRIBUTING guidelines](https://github.com/eProsima/Fast-DDS/blob/master/CONTRIBUTING.md#unofficial-support).

Also, you might be interested in contacting [Fast DDS commercial support](https://github.com/eProsima/Fast-DDS#getting-help) if you need some further help with your use case.
